src/battle_util.c

```diff
@@ -7907,7 +7905,7 @@ u8 ItemBattleEffects(u8 caseID, u32 battler, bool32 moveTurn)
         switch (battlerHoldEffect)
         {
         case HOLD_EFFECT_TOXIC_ORB:
-            if (CanBePoisoned(battler, battler))
+            if (CanBePoisoned(battler, battler, ABILITY_NONE))
```

Comment:
Should be `GetBattlerAbility` here instead of `ABILITY_NONE`

Reply:
That is poison through toxic orb. I believe abilities from user are ignored in this case

Reply:
It's not the case, a pokemon with Immunity cannot be poisoned with Toxic Orb.

Reply:
adeed